packages/suite/src/actions/wallet/sendFormSpecific/rippleActions.ts

```diff
@@ -1,12 +1,74 @@
 import TrezorConnect from 'trezor-connect';
+import { calculateTotal, getOutput, calculateMax } from '@wallet-utils/sendFormUtils';
+import { networkAmountToSatoshi } from '@wallet-utils/accountUtils';
+import { XRP_FLAG } from '@wallet-constants/sendForm';
 import { SEND } from '@wallet-actions/constants';
+import Bignumber from 'bignumber.js';
+import { PrecomposedTransactionXrp } from '@wallet-types/sendForm';
 import { NOTIFICATION } from '@suite-actions/constants';
 import { Dispatch, GetState } from '@suite-types';
 
-export interface SendFormRippleActions {
-    type: typeof SEND.XRP_HANDLE_DESTINATION_TAG_CHANGE;
-    destinationTag: string;
-}
+export type SendFormRippleActions =
+    | { type: typeof SEND.XRP_HANDLE_DESTINATION_TAG_CHANGE; destinationTag: string }
+    | { type: typeof SEND.XRP_PRECOMPOSED_TX; payload: PrecomposedTransactionXrp };
+
+/*
+    Compose xrp transaction
+ */
+export const compose = () => async (dispatch: Dispatch, getState: GetState) => {
+    const { send } = getState().wallet;
+    const { account } = getState().wallet.selectedAccount;
+    if (!send || !account) return null;
+
+    const output = getOutput(send.outputs, 0);
+    const amountInSatoshi = networkAmountToSatoshi(
+        output.amount.value || '0',
+        account.symbol,
+    ).toString();
+    const { availableBalance } = account;
+    const feeInSatoshi = send.selectedFee.value;
+    let tx;
+    const totalSpentBig = new Bignumber(calculateTotal(amountInSatoshi || '0', feeInSatoshi));
```

Comment:
amountInSatoshi shouldn't be undefined, you have already declare fallback value in line 25 `output.amount.value || '0'` so there is no need to do the same, right?